dialogy/plugins/text/intent_entity_mutator/__init__.py

```diff
@@ -0,0 +1,273 @@
+"""
+    IntentEntityMutatorPlugin
+"""
+
+from typing import Any, List, Optional, Dict, Iterable, Union, Callable
+
+from dialogy.base import Input, Output, Plugin, Guard
+from dialogy.types import BaseEntity, Intent
+
+from dialogy.plugins.text.intent_entity_mutator.actions_on_primitives import (
+    contain_digits,
+    is_number_absent,
+)
+
+transcript_functions_map: Dict[str, Callable[[List[str]], int]] = {
+    "is_number_absent": is_number_absent,
+    "contain_digits": contain_digits,
+}
+
+
+class IntentEntityMutatorPlugin(Plugin):
+    def __init__(
+        self,
+        dest: Optional[str] = None,
+        guards: Optional[List[Guard]] = None,
+        rules: Optional[Dict[str, List[Any]]] = None,
```

Comment:
Why not make `rules` a necessary argument of the plugin? You don't need the `if` block in that case.

Reply:
Done